cmd/org/clusters.go

```diff
@@ -74,70 +90,77 @@ func init() {
 	AddOutputFlag(flags)
 }
 
-func SearchClusters(cmd *cobra.Command, args []string) error {
-
-	var err error
-	if !hasOrgId(args) && !isAWSProfileSearch() {
-		err = fmt.Errorf("specify either org-id or --aws-profile,--aws-account-id arguments")
+func SearchClusters(orgId string, status string) ([]*v1.Subscription, error) {
+	if orgId == "" && !isAWSProfileSearch() {
+		return nil, fmt.Errorf("specify either org-id or --aws-profile,--aws-account-id arguments")
 	}
-	if hasOrgId(args) {
-		err = searchclustersByOrg(cmd, args[0])
+
+	if orgId != "" && isAWSProfileSearch() {
+		return nil, fmt.Errorf("specify either an org id argument or --aws-profile, --aws-account-id arguments")
 	}
+
 	if isAWSProfileSearch() {
-		err = searchClustersByAWSProfile(cmd)
+		orgIdFromAws, err := getOrganizationIdFromAWSProfile()
+		orgId = *orgIdFromAws
+		if err != nil {
+			return nil, fmt.Errorf("failed to get org ID from AWS profile: %w", err)
+		}
```

Comment:
**issue:** We're dereferencing the result (possibly a nil ptr) before checking `err != nil`

Reply:
Great catch on this! Fixing now.